glean-core/csharp/Glean/LibGleanFFI.cs

```diff
@@ -8,6 +8,63 @@
 
 namespace Mozilla.Glean.FFI
 {
+    /// <summary>
+    /// Result values of attempted ping uploads encoded for FFI use.
+    /// They are defined in `glean-core/src/upload/result.rs` and re-defined for use in Kotlin here.
+    /// 
+    /// NOTE:
+    /// THEY MUST BE THE SAME ACROSS BOTH FILES!
+    /// </summary>
+    internal enum Constants : int
+    {
+        // A recoverable error.
+        UPLOAD_RESULT_RECOVERABLE = 0x1,
+
+        // An unrecoverable error.
+        UPLOAD_RESULT_UNRECOVERABLE = 0x2,
+
+        // A HTTP response code.
+        UPLOAD_RESULT_HTTP_STATUS = 0x8000
+    }
+
+    /// <summary>
+    /// Rust represents the upload task as an Enum
+    /// and to go through the FFI that gets transformed into a tagged union.
+    /// Each variant is represented as an 8-bit unsigned integer.
+    ///
+    /// This *MUST* have the same order as the variants in `glean-core/ffi/src/upload.rs`.
+    /// </summary>
+    enum UploadTaskTag : int
+    {
+        Upload,
+        Wait,
+        Done
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct FfiUploadTaskBody
+    {
+        public byte tag;
+        public IntPtr documentId;
+        public IntPtr path;
+        public int bodyLen;
+        public IntPtr body;
```

Comment:
nit: It looks ok to me. But if you wanna match to `UploadBody.kt`, there is supposed to have a RustBuffer struct to contain these.

Reply:
In Kotlin, we copied over that implementation from Application Services. In this case, we don't have any reference implementation available, so we're fine to leave this as it is. There it was more for convenience, than for need.